backend/layers/processing/upload_failures/app.py

```diff
@@ -2,30 +2,69 @@
 import json
 import logging
 import os
+from typing import Optional
 
 from backend.common.utils.aws import delete_many_from_s3
-from backend.common.utils.json import CustomJSONEncoder
-from backend.common.utils.result_notification import format_failed_batch_issue_slack_alert, notify_slack
+from backend.common.utils.result_notification import aws_batch_job_url_fmt_str, aws_sfn_url_fmt_str, notify_slack
 from backend.layers.common.entities import DatasetProcessingStatus, DatasetStatusKey, DatasetVersionId
 from backend.portal.api.providers import get_business_logic
 
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
 
 def handle_failure(event: dict, context) -> None:
-    dataset_id = event["dataset_id"]
-    trigger_slack_notification(dataset_id)
-    update_dataset_processing_status_to_failed(dataset_id, event["error"]["Cause"])
+    (
+        dataset_id,
+        collection_version_id,
+        error_step_name,
+        error_job_id,
+        error_aws_regions,
+        error_cause,
+        execution_arn,
+    ) = parse_event(event)
+    trigger_slack_notification(
```

Comment:
worth putting this in a try/except clause so as to ensure the dataset processing status is still updated even if the slack notification generates an unhandled error?